lib/src/utils/autocomplete_manager.dart

```diff
@@ -0,0 +1,47 @@
+import 'dart:async';
+
+import 'package:meta/meta.dart';
+import 'autocompleter.dart';
+
+/// Manager that returns the suggestions for the latest input.
+///
+/// Typical use case: the user types one character (which triggers a call to
+/// suggestions), then another character (which triggers another call).
+/// What if the second call is much faster than the first one, for server or
+/// connection reasons? The autocomplete widget will get the second suggestions,
+/// then the first suggestions will override them.
+/// And the user should get the suggestions that match the latest input.
+class AutocompleteManager implements Autocompleter {
```

Comment:
I wonder if you shouldn't provide a `clear` method, to allow the client to clear the cache, in the case where the same Manager is used for multiple queries